# HG changeset patch
# User Rik <rik@octave.org>
# Date 1419905274 28800
#      Mon Dec 29 18:07:54 2014 -0800
# Node ID 9552138fe6f580ddcc2bd9a2b855c42e283abf9b
# Parent  51a1d1164449160403c38890fc1c88502feea550
Fix line color when using plot3 (bug #39585, bug #36926).

* __go_draw_axes__.m: For line objects with 3 dimensions, set hidden_removal to
false.

diff --git a/scripts/plot/util/private/__go_draw_axes__.m b/scripts/plot/util/private/__go_draw_axes__.m
--- a/scripts/plot/util/private/__go_draw_axes__.m
+++ b/scripts/plot/util/private/__go_draw_axes__.m
@@ -585,16 +585,17 @@ function __go_draw_axes__ (h, plot_strea
           if (! isempty (obj.zdata))
             zdat = obj.zdata(:);
           else
             zdat = zeros (size (xdat));
           endif
           data{data_idx} = [xdat, ydat, zdat]';
           usingclause{data_idx} = sprintf ("record=%d using ($1):($2):($3)",
                                            numel (xdat));
+          hidden_removal = false;
           ## fputs (plot_stream, "set parametric;\n");
         else
           xdat = obj.xdata(:);
           ydat = obj.ydata(:);
           data{data_idx} = [xdat, ydat]';
           usingclause{data_idx} = ...
             sprintf ("record=%d using ($1):($2) axes %s%s",
                      rows (xdat), xaxisloc_using, yaxisloc_using);
